File: src/pokerbots/spam_framework/GameState.java
package pokerbots.spam_framework;

public class GameState {
	
	// Street enum values
	public static final int PREGAME = 0;
	public static final int PREFLOP = 1;
	public static final int FLOP = 2;
	public static final int TURN = 3;
	public static final int RIVER = 4;
	public static final int GAMEOVER = 5;
	
	// Attributes
	private int numStartingChips;	// number of chips
	private int numMyChips;			// number of chips left in player's stack
	private int numVillianChips;	// number of chips left in other player's stack
	private int bigBlindAmount; 	// number of chips for a big blind
	private int numHandsLeft;		// number of hands left
	private int currentHandID; 		// ID of the current hand
	private double handDuration;	// duration allowed for each match
	private double timeLeft;		// time left before match ends
	private boolean isDealer; 		// whether or not we are the dealer
	private Card card1, card2;		// the two cards the player has
	private int currentStreet;		// the current street of the game

	private PlayerHistory selfHistory;	  // stats on our moves
	private PlayerHistory villainHistory; // stats on previous villain moves
	
	
	
	// ----- GETTERS AND SETTERS --------------------------------------------
	
	public int getNumMyChips() {
		return numMyChips;
	}

	public void setNumMyChips(int numMyChips) {
		this.numMyChips = numMyChips;
	}

	public int getNumVillianChips() {
		return numVillianChips;
	}

	public void setNumVillianChips(int numVillianChips) {
		this.numVillianChips = numVillianChips;
	}

	public int getBigBlindAmount() {
		return bigBlindAmount;
	}

	public void setBigBlindAmount(int bigBlindAmount) {
		this.bigBlindAmount = bigBlindAmount;
	}

	public int getNumHandsLeft() {
		return numHandsLeft;
	}

	public void setNumHandsLeft(int numHandsLeft) {
		this.numHandsLeft = numHandsLeft;
	}

	public double getHandDuration() {
		return handDuration;
	}

	public void setHandDuration(double handDuration) {
		this.handDuration = handDuration;
	}

	public boolean isDealer() {
		return isDealer;
	}

	public void setDealer(boolean isDealer) {
		this.isDealer = isDealer;
	}

	public Card getCard2() {
		return card2;
	}

	public void setCard2(Card card2) {
		this.card2 = card2;
	}

	public Card getCard1() {
		return card1;
	}

	public void setCard1(Card card1) {
		this.card1 = card1;
	}

	public int getCurrentHandID() {
		return currentHandID;
	}

	public void setCurrentHandID(int currentHandID) {
		this.currentHandID = currentHandID;
	}

	public int getNumStartingChips() {
		return numStartingChips;
	}

	public void setNumStartingChips(int numStartingChips) {
		this.numStartingChips = numStartingChips;
	}

	public double getTimeLeft() {
		return timeLeft;
	}

	public void setTimeLeft(double timeLeft) {
		this.timeLeft = timeLeft;
	}

	public PlayerHistory getVillainHistory() {
		return villainHistory;
	}

	public void setVillainHistory(PlayerHistory villainHistory) {
		this.villainHistory = villainHistory;
	}

	public PlayerHistory getSelfHistory() {
		return selfHistory;
	}

	public void setSelfHistory(PlayerHistory selfHistory) {
		this.selfHistory = selfHistory;
	}

	public int getCurrentStreet() {
		return currentStreet;
	}

	public void setCurrentStreet(int currentStreet) {
		this.currentStreet = currentStreet;
	}
	
	
}

File: spambot/SPAMFramework/GameState.cpp
#include "GameState.h"

	GameState::GameState::GameState() {
		m_self_history = new PlayerHistory();
		m_villain_history = new PlayerHistory();
	}

	int GameState::GetMyStackSize(){
		return m_stack_size;
	}

	void GameState::SetMyStackSize(int num_my_chips){
		m_stack_size = num_my_chips;
	}

	int GameState::GetVillainStackSize(){
		return m_villain_stack_size;
	}

	void GameState::SetVillainStackSize(int num_villian_chips){
		m_villain_stack_size = num_villian_chips;
	}

	int GameState::GetBigBlindAmount(){
		return m_big_blind_amount;
	}

	void GameState::SetBigBlindAmount(int big_blind_amount){
		m_big_blind_amount = big_blind_amount;
	}

	int GameState::GetNumHandsLeft(){
		return m_num_hands_left;
	}

	void GameState::SetNumHandsLeft(int num_hands_left){
		m_num_hands_left = num_hands_left;
	}

	double GameState::GetHandDuration(){
		return m_hand_duration;
	}

	void GameState::SetHandDuration(double hand_duration){
		m_hand_duration = hand_duration;
	}

	bool GameState::IsDealer(){
		return m_is_dealer;
	}

	void GameState::SetDealer(bool is_dealer){
		m_is_dealer = is_dealer;
	}

	Card* GameState::GetCard2(){
		return m_card2;
	}

	void GameState::SetCard2(Card *card2){
		m_card2 = card2;
	}

	Card* GameState::GetCard1(){
		return m_card1;
	}

	void GameState::SetCard1(Card *card1){
		m_card1 = card1;
	}

	int GameState::GetCurrentHandID(){
		return m_current_hand_id;
	}

	void GameState::SetCurrentHandID(int current_hand_id){
		m_current_hand_id = current_hand_id;
	}

	int GameState::GetNumStartingChips(){
		return m_num_starting_chips;
	}

	void GameState::SetNumStartingChips(int num_starting_chips){
		m_num_starting_chips = num_starting_chips;
	}

	double GameState::GetTimeLeft(){
		return m_time_left;
	}

	void GameState::SetTimeLeft(double time_left){
		m_time_left = time_left;
	}

	PlayerHistory* GameState::GetVillainHistory(){
		return m_villain_history;
	}

	void GameState::SetVillainHistory(PlayerHistory *villain_history){
		m_villain_history = villain_history;
	}

	PlayerHistory* GameState::GetSelfHistory(){
		return m_self_history;
	}

	void GameState::SetSelfHistory(PlayerHistory *self_history){
		m_self_history = self_history;
	}

	int GameState::GetCurrentStreet(){
		return m_current_street;
	}

	void GameState::SetCurrentStreet(int current_street){
		m_current_street = current_street;
	}

	int GameState::GetPotSize(){
		return m_pot_size;
	}

	void GameState::SetPotSize(int pot_size){
		m_pot_size = pot_size;
	}

	std::vector<std::string> GameState::GetLegalActions(){
		return m_legal_actions;
	}

	void GameState::SetLegalActions(std::vector<std::string> legal_actions){
		m_legal_actions = legal_actions;
	}

	std::vector<Card*> GameState::GetBoardCards(){
		return m_board_cards;
	}

	void GameState::SetBoardCards(std::vector<Card*> board_cards){
		m_board_cards = board_cards;
	}
